### PR TITLE
Angular sample - update to v15

### PR DESCRIPTION
**Pull request by andygup**
Update Angular sample from 14 to 15. Notable items:
* The upgrade script deleted our `.browserslistrc` and said it duplicated the default build settings.
* The upgrade script made a few adjustments to the Karma tests. Those still run okay.
* Added Known issue note in the README about Angular 15 changes to tsconfig and our requirements for setting `useDefineForClassFields`. 
* Build times increased significantly between Angular 14 and Angular 15. Angular did make changes under the hood for ESBuild. Didn't find any obvious quick fixes. There's a [somewhat related conversation](https://github.com/angular/angular-cli/issues/24163) on the Angular CLI repo about configuring the builder.

### CODE REVIEW

File: esm-samples/jsapi-angular-cli/README.md [hunk 1]
```diff
@@ -4,14 +4,16 @@ This repo demonstrates how to use [@arcgis/core](https://www.npmjs.com/package/@
 
 ---
 ## Known issues
+* Upgrading Angular 15 automatically sets the TypeScript `target` to `ES2022` and `useDefineForClassFields` to `false`. The ArcGIS JS API requires that `useDefinedForClassFields` be set to `false` when using
```
Comment (odoe) [inline esm-samples/jsapi-angular-cli/README.md:7]: This sounded a little confusing to me. Angular set this flag to `false`, and we also require it to be `false`? Maybe change the wording a bit to highlight that this change is in line with our requirements?

**Review by odoe — APPROVED** [review]
:+1: Just one comment an a sentence that looked odd to me, but rest looks good!

File: esm-samples/jsapi-angular-cli/tsconfig.json [hunk 2]
```diff
@@ -16,12 +16,13 @@
...
     "importHelpers": true,
-    "target": "es2020",
+    "target": "ES2022",
     "module": "es2020",
     "lib": [
       "es2020",
       "dom"
-    ]
+    ],
+    "useDefineForClassFields": false
```
Comment (dasa) [inline esm-samples/jsapi-angular-cli/tsconfig.json:25]: - should the lib be updated too match the target?
- do we need to explicitly set `useDefineForClassFields`, or is this to benefit older versions?

**Review by dasa — APPROVED** [review]

File: esm-samples/jsapi-angular-cli/README.md (hunk 1 above)
Comment (andygup) [reply esm-samples/jsapi-angular-cli/README.md]: Good point, I updated it and removed the Angular 15 reference.

**Pull request merged**